### PR TITLE
*: SubSelect can be used as select statement

### PR DESCRIPTION
**Pull request by shenli**
Fix https://github.com/pingcap/tidb/issues/320
@siddontang

### CODE REVIEW

File: parser/parser.y [hunk 1]
```diff
@@ -3437,6 +3437,11 @@ Statement:
 |	UnionStmt
 |	UpdateStmt
 |	UseStmt
+|	SubSelect
```
Comment (siddontang) [inline parser/parser.y]: I think here may be `'(' SelectStmt ')', not SubSelect, in MySQL, SubSelect is not the same as select. 
And you can see the MySQL parser that it uses select too, not sub select.

Comment (shenli) [inline parser/parser.y]: I tried to add '(' SelectStmt ')' into SelectStmt rule, but I got unresolveable confliction.

**Comment by siddontang** [comment]
LGTM
But we must realize that select is not the same as sub select, maybe we will meet some pitfalls later. :-)

File: tidb.go [hunk 2]
```diff
@@ -257,6 +257,14 @@ func trimSQL(sql string) string {
 		}
 		break
 	}
+	// Trim leading '('. For `(select 1);` is also a query.
```
Comment (qiuyesuifeng) [inline tidb.go]: If trim '(', any need to trim ')' suffix?

Comment (shenli) [inline tidb.go]: No, We judge if the statement is a query only by the prefix

Comment (qiuyesuifeng) [inline tidb.go]: Then sql will be `select 1);`
Seems like an invalid sql?

Comment (shenli) [inline tidb.go]: This is only used to check if sql is query in interpreter. And the trimed sql text will not be used outside this function.

Comment (shenli) [inline tidb.go]: PTAL @qiuyesuifeng

**Pull request merged**